backend/apis/clients/follow-ups/create-follow-up.api.js

```diff
@@ -0,0 +1,94 @@
+const { app, databaseError, pool, invalidRequest } = require("../../../server");
+const mysql = require("mysql");
+const {
+  checkValid,
+  validId,
+  validDate,
+  validTime,
+  validArray,
+  validDateTime,
+  nullableValidDateTime,
+} = require("../../utils/validation-utils");
+const fs = require("fs");
+const path = require("path");
+const followUpsSql = fs.readFileSync(
+  path.resolve(__dirname, "./create-follow-up.sql"),
+  "utf-8"
+);
+const getFollowUpSql = fs.readFileSync(
+  path.resolve(__dirname, "./get-follow-up.sql"),
+  "utf-8"
+);
+
+app.post("/api/clients/:clientId/follow-ups", (req, res) => {
+  const user = req.session.passport.user;
+
+  const validationErrors = [
+    ...checkValid(req.params, validId("clientId")),
+    ...checkValid(req.body, validArray("serviceIds", validId)),
+    ...checkValid(req.body, validDateTime("dateOfContact")),
+    ...checkValid(req.body, nullableValidDateTime("appointmenetDate")),
+  ];
+
+  const { clientId } = req.params;
+
+  const {
+    serviceIds,
+    title,
+    description,
+    dateOfContact,
+    appointmentDate,
+  } = req.body;
+
+  if (validationErrors.length > 0) {
+    return invalidRequest(res, validationErrors);
+  }
+
+  const insertFollowUpSql = mysql.format(followUpsSql, [
+    clientId,
+    title,
+    description,
+    dateOfContact,
+    appointmentDate,
+    user.id,
+    user.id,
+  ]);
+
+  pool.query(insertFollowUpSql, (err, insertResult) => {
+    if (err) {
+      return databaseError(req, res, err);
+    }
+
+    let query = "";
+
+    serviceIds.forEach((id) => {
+      query += `
+      INSERT INTO followUpServices (serviceId, followUpId) VALUES (${id}, ${insertResult.insertId});
+      `;
+    });
+
+    pool.query(query, (err, joinResult) => {
+      if (err) {
+        return databaseError(req, res, err);
+      }
+
+      formattedFollowUpResponse(insertResult.insertId, (err, result) => {
+        if (err) {
+          return databaseError(err);
+        }
+        res.send(result);
+      });
+    });
+  });
+});
+
+function formattedFollowUpResponse(id, errBack) {
```

Comment:
Generally when using JSON_ARRAYAGG and maybe also JSON_OBJECT you have to call `JSON.parse()` on the returned fields, because the mysql driver gives you the data as a string instead of an array or object. Have you confirmed whether that is the case for the `serviceIds`, `createdBy`, and `modifiedBy` properties?

Reply:
Ah my bad, I thought that was handled on the front end! Will parse on server end!